src/gui/elems/basics/menuButton.cpp

```diff
@@ -0,0 +1,93 @@
+/* -----------------------------------------------------------------------------
+ *
+ * Giada - Your Hardcore Loopmachine
+ *
+ * -----------------------------------------------------------------------------
+ *
+ * Copyright (C) 2010-2020 Giovanni A. Zuliani | Monocasual
+ *
+ * This file is part of Giada - Your Hardcore Loopmachine.
+ *
+ * Giada - Your Hardcore Loopmachine is free software: you can
+ * redistribute it and/or modify it under the terms of the GNU General
+ * Public License as published by the Free Software Foundation, either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * Giada - Your Hardcore Loopmachine is distributed in the hope that it
+ * will be useful, but WITHOUT ANY WARRANTY; without even the implied
+ * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Giada - Your Hardcore Loopmachine. If not, see
+ * <http://www.gnu.org/licenses/>.
+ *
+ * -------------------------------------------------------------------------- */
+
+
+#include <FL/fl_draw.H>
+#include "core/const.h"
+#include "utils/gui.h"
+#include "menuButton.h"
+
+
+namespace giada {
+namespace v
+{
+
+geMenuButton::geMenuButton(int x, int y, int w, int h, const char* l,
+							const char* txt) 
+: Fl_Menu_Button(x, y, w, h, l),
+fg_text(txt)
```

Comment:
`txt` is `nullptr` by default. You should check if `txt` is `nullptr`. It would probably better to change the type of `txt` to a `const std::string &` since you save it as a string anyways.

Reply:
Good point, I'll see into that.